### ISSUE TITLE
Existing ConvNeXt Doesn't Train on GPU

### ISSUE THREAD

**Issue by DavidLandup0** [keras-team/keras-cv#1235]
Discovered while trying to benchmack ConvNeXtV2 vs ConvNeXt.

The existing ConvNeXt implementation doesn't train in `keras.applications` and `keras_cv`. The loss is going down, but the accuracy stays flat (at ~random guessing). This feels like a numerical instability issue, even though it's trained at full-floats.
Might be an issue with GPUs only?

Example colab for reproduction:
https://colab.research.google.com/drive/1-fDt3XDm1WmUGVo-29Ap9-BIiTH8sZrQ

EffNet trains on the same setup. Uses KCV preprocessing layers and TFDS.

@sayakpaul @tanzhenyu @LukeWood @ianstenbit

**Comment by sayakpaul** [comment]
I think it's a little unfair to make this comparison. This is especially because ConvNeXt was evolved to a CNN-based architecture but with motivations from Transformers. Many augmentation transformations and training hyperparameters like optimizer, learning rate that work well for other CNN architectures (like EfficientNetV2) may not extend to ConvNeXt. 

One immediate thing is SGD. ConvNeXt was originally intentionally trained with Transformers-like hyperparameters (Adam for example). 

With these things in mind, when I changed the learning rate to 0.001 it seems to train as expected (although not expected performance). Tweaking the hyperparameters is likely the key here :) 

Here's my Colab Gist: https://colab.research.google.com/gist/sayakpaul/15b86f1ba5017a3202e1c62584043197/convnext.ipynb

**Comment by DavidLandup0** [comment]
My bad. I tried AdamW (1.5e-4), SGD (0.01), Adam (0.001) and totally removing augs and the results were the same, but it was probably an LR too high 🤦

**Comment by DavidLandup0** [comment]
I made the comparison primarily to check whether the input pipeline works, not to measure performances. Should've been more clear about that. Thanks for weighing in!

**Comment by DavidLandup0** [comment]
Just realized what the issue was. I referenced the training setup from the paper (such as AdamW with 1e-4 lr, etc.) knowing that the setups will necessarily differ, but forgot to scale the LR with the batch size. They used 4096 while I used 32, so my LR was *way* higher than it should've been across *all* optimizers.

Seeing the same result across AdamW, Adam and SGD made me think it was a GPU issue, but I jumped to the conclusion. Sorry for the ping and thanks for the fast response!
